the-unravelling/Assets/Scripts/InventoryUIBehaviour.cs

```diff
@@ -0,0 +1,58 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public delegate void OnClickInventory(in ItemData item);
+
+/// <summary>
+/// The inventory UI Behaviour drives the UI for the inventory, by handling opening and closing the UI, as well as keeping the list of items up to data.
+/// </summary>
+public class InventoryUIBehaviour : MonoBehaviour {
+    // Prefab for the item to be instantiated
+    public ItemPreview itemPrefab;
+
+    // The panel that is part of the UI
+    public GameObject panel;
+
+    // Player callback, called when the inventory ui closes
+    private OnClickInventory callback;
+
+    /// <summary>
+    /// Activate the UI, and populate the list of items.
+    /// </summary>
+    /// <param name="inventory">Inventory (list of items)</param>
+    /// <param name="click">Callback called when the inventory is closed again</param>
+    public void OnActivate(in Inventory inventory, OnClickInventory click) {
+        callback = click;
+
+        // Activate the UI
+        gameObject.SetActive(true);
+
+        // Destroy all previous elements
+        for (var i = 0; i < panel.transform.childCount; i++) {
+            Destroy(panel.transform.GetChild(i).gameObject);
```

Comment:
Burde kanskje ha dette på CloseInventory() istedenfor

Reply:
Not a bad idea